xtc/src/ledger.rs

```diff
@@ -71,6 +81,27 @@ impl Ledger {
     }
 }
 
+#[derive(CandidType, Clone)]
```

Comment:
other option would be
```rust
enum APIError {
    InsufficientBalance(&'static str),
    Unknown(&'static str),
}
```

However, in that case retrieving the error message would require match expression, unlike with the proposed solution.